### ISSUE TITLE
HttpException returns an ugly message.

### ISSUE THREAD

**Issue by VictorioBerra** [Dotnet-Boxed/Framework#64]
HttpException calls base(message), if message is null you get something ugly like:

`Exception of type 'XYZ+HttpException' was thrown.`

This is the exposed here https://github.com/Dotnet-Boxed/Framework/blob/master/Source/Boxed.AspNetCore/Middleware/HttpExceptionMiddleware.cs#L29

## Also:

It is important to note that ReasonPhrase is not a thing in HTTP/2. Why not have something like this?

```csharp
            if (exception is HttpException re)
            {
                context.Response.StatusCode = (int)re.StatusCode;
                var result = JsonConvert.SerializeObject(new
                {
                    errors = re.Message
                });
                await context.Response.WriteAsync(result);
            }
```

and then in HttpException `base(message ?? string.Empty)`

**Comment by RehanSaeed** [comment]
Honestly, I'm not a fan of using the message anywhere. I only put reason phrase in there because that is what MVC5 was doing. Exceptions can contain sensitive data and I don't think they should ever be exposed to the clients.

**Comment by VictorioBerra** [comment]
@cannehag @RehanSaeed Yeah, I wish .NET would offer some better alternatives to throwing a custom exception and then using middleware to catch and expose it. That is such an easy way to handle things. Until there is an easier way a lot of people will want to go down this path.

Jonas has a point that exceptions are a first-class citizen in ASPNET Core.

> force you to return errors alongside the result of your function

@RehanSaeed Do you have an articles or examples for doing it the "Rust way" in C#?

I wouldn't want to have to create a DTO for every single one of my methods that has an `Errors` property. That would be too much work.

**Comment by RehanSaeed** [comment]
> @RehanSaeed Do you have an articles or examples for doing it the "Rust way" in C#?

Use a custom class containing your success and error properties. Or you can use modern tuples to save on having to write extra classes.

**Comment by RehanSaeed** [comment]
Released
